client/domain/CorrectnessTestObjectFactory.js

```diff
@@ -24,13 +24,21 @@ tie.factory('CorrectnessTestObjectFactory', [
       this._input = correctnessTestDict.input;
       this._allowedOutputs = correctnessTestDict.allowedOutputs;
       this._message = correctnessTestDict.message;
+      this._tag = correctnessTestDict.tag;
     };
 
     // Instance methods.
     CorrectnessTest.prototype.getInput = function() {
       return this._input;
     };
 
+    CorrectnessTest.prototype.getStringifiedInput = function() {
+      if (!this._stringifiedInput) {
+        this._stringifiedInput = JSON.stringify(this._input);
```

Comment:
Let's either always do it at object creation time, or always compute it on the fly. I'm fine either way.